### PR TITLE
Add Create Product API

### PR DESCRIPTION
**Pull request by ahmedmq**
- Define global exception handler
- Define and Handle bean validation exception
- Handle product code already exists

### CODE REVIEW

File: src/main/java/com/sivalabs/bookstore/catalog/config/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException ex, HttpHeaders headers, HttpStatusCode status, WebRequest request) {
+        String errorMessage = ex.getBindingResult().getAllErrors().stream()
+                .map(err -> err.unwrap(ConstraintViolation.class))
+                .map(err -> String.format("'%s' %s", err.getPropertyPath(), err.getMessage()))
+                .collect(Collectors.joining(","));
+        ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, errorMessage);
+        problemDetail.setTitle("Invalid Request");
+        problemDetail.setType(URI.create("https://api.bookmarks.com/errors/bad-request"));
+        problemDetail.setProperty("timestamp", Instant.now());
```
Comment (rajadilipkolli) [inline src/main/java/com/sivalabs/bookstore/catalog/config/GlobalExceptionHandler.java:31]: Can you please add type of error as well to distinguish easily

problemDetail.setProperty("errorCategory", "Validation");

Comment (ahmedmq) [reply src/main/java/com/sivalabs/bookstore/catalog/config/GlobalExceptionHandler.java:31]: Done 👍🏾

**Review by rajadilipkolli — APPROVED** [review]
Approved

**Comment by sivaprasadreddy** [comment]
@ahmedmq Please rebase your changes to resolve the conflicts.

**Pull request merged**